### ISSUE TITLE
`node` subcommands that create new nodes should hide the `node_name` default value

### ISSUE THREAD

**Issue by adrianbenavides** [build-trust/ockam#3231]
## Current

`node` subcommands have `node_name` as the first argument. The commands that create new nodes assign a default random value when this argument is not set. This value should be hidden.

Currently, this is only done correctly in the `node create` command.

## Desired

The following commands should have the `hide_default_value = true` attribute in the `node_name` argument (take the `node create` command as reference):

- `node start` 
- `node stop`

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

**Comment by hargut** [comment]
Hi @adrianbenavides,

the ticket is not really clear how this should be handled. Just hiding the argument leads to issues, as the randomly generated nodenames are not available during start/stop. Config is missing or running node is missing when stopping/starting.

How should this be solved? Possible approaches could be:

Start:
- checking existing configuration and using the one existing node in case there is just one, exiting in case there are multiple
- implicitly creating a new node with random name and starting it

Stop:
- checking on stop if a node is running, and in case it is just one stopping this node and exiting in case multiple are running

**Comment by adrianbenavides** [comment]
Let me give you more details. When you run `ockam node start --help` you will see the following:

```bash
Usage:
    ockam node start [OPTIONS] [NODE_NAME]

Arguments:
  [NODE_NAME]  Name of the node [default: 2f4f983b]
```

Now, if you add the `hide_default_value = true` attribute in the `node_name` argument, you will hide the `[default: 2f4f983b]` part only, not the argument itself:

```bash
Usage:
    ockam node start [OPTIONS] [NODE_NAME]

Arguments:
  [NODE_NAME]  Name of the node
```

We want to hide that text because it's not really a default value, as it's different every time you run the command.

Let me know if that helped!

**Comment by hargut** [comment]
@adrianbenavides 
That part was clear, but it looks like the headline of the ticket got me to over-think the task.

The indication that "subcommands that create new nodes" brought me into the direction of assuming the argument should as well be optional, which raised the questions.

Sorry for complicating the request. The PR to just hide the value is linked.

**Comment by elpiel** [comment]
Should be closed with #3621?

**Comment by adrianbenavides** [comment]
Correct, thanks again!
